crates/iota-mainnet-unlocks/src/bin/generate_aggregated_data.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::BTreeMap,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use anyhow::{Context, Result};
+use chrono::{DateTime, Utc};
+use csv::{Reader, Writer};
+use iota_mainnet_unlocks::store::{INPUT_FILE as OUTPUT_FILE, StillLockedEntry};
+use regex::Regex;
+use tempfile::{TempDir, tempdir};
+
+// Folders in the raw data repository that contain the CSV files.
+const FOLDERS: &[&str] = &[
+    "Assembly_IF_Members",
+    "Assembly_Investors",
+    "IOTA_Airdrop",
+    "IOTA_Foundation",
+    "New_Investors",
+    "TEA",
+    "Treasury_DAO",
+    "UAE",
+];
+
+/// Clones the repository containing raw data into a temporary directory.
+fn clone_repo(tmp_dir: &TempDir) -> Result<PathBuf> {
+    let repo_path = tmp_dir.path().join("new_supply");
+
+    let status = Command::new("git")
+        .args([
+            "clone",
+            "--depth",
+            "1",
+            "https://github.com/iotaledger/new_supply.git",
+        ])
+        .arg(&repo_path)
+        .status()
+        .context("failed to execute `git clone`")?;
+
+    if !status.success() {
+        anyhow::bail!("`git clone` failed with exit status: {}", status);
+    }
+
+    Ok(repo_path)
+}
+
+/// Reads and aggregates the CSV unlock data from the cloned repository.
+/// Returns a BTreeMap keyed by unlock date (as a String) with the aggregated
+/// token amount (in nano-units).
+fn aggregate_unlocks(repo_path: &Path) -> Result<BTreeMap<String, u64>> {
+    let mut locked_by_date: BTreeMap<String, u64> = BTreeMap::new();
+
+    for folder in FOLDERS {
+        let csv_path = repo_path.join(folder).join("summary.csv");
+        println!("Processing file: {:?}", csv_path);
+
+        let mut rdr = Reader::from_path(&csv_path)
+            .with_context(|| format!("failed to open CSV file: {csv_path:?}"))?;
+
+        // Iterate over CSV records (header is skipped automatically).
+        for result in rdr.records() {
+            let record = result?;
+            if record.len() < 2 {
+                eprintln!("Skipping invalid record: {:?}", record);
```

Comment:
did you actually encounter one? Sounds important enough to not ignore it

Reply:
Fixed in https://github.com/iotaledger/iota/pull/6123/commits/1a16ba645060ca55ffe5edd134c95a8cc66570ac